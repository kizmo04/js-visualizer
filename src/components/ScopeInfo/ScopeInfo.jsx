import React, { Component } from "react";
import styles from "./ScopeInfo.module.scss";
import _ from "lodash";

class ScopeInfo extends Component {
  render() {
    const { scopeHistory, currentScope, operationType } = this.props;
    return (
      <div className={`${styles.logBox}`}>
        <h2 className={`${styles.font} ${styles.underline} has-text-success operation-type`}>
          {operationType ? `${operationType}` : null}
        </h2>
        {_.map([...scopeHistory, currentScope], (scope, index) => (
          <div key={index} className={`${styles.scopeBox}`}>
            {_.map(scope, (info, key) => {
              if (key === "scopeName") {
                return (
                  <h2 key={index + key} className={`${styles.font} ${styles.closure} has-text-success closure`}>
                    {info.value} Closure
                  </h2>
                );
              } else {
                return (
                  <p key={index + key} ref className={`${styles.font} ${info.highlight ? 'has-background-danger has-text-warning' : 'has-text-success'} property`}>{`${key}: ${
                    info.value
                  }`}</p>
                );
              }
            })}
          </div>
        ))}
      </div>
    );
  }
}

export default ScopeInfo;
