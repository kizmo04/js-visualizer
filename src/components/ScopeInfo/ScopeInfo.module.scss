.logBox {
  min-height: 30vh;
  margin-left: 40px;
  overflow: scroll;
}

.font {
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  font-family: monospace, BlinkMacSystemFont, -apple-system, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", "Helvetica", "Arial", sans-serif;
}

.underline {
  margin-bottom: 2rem;
}

.scopeBox {
  float: left;
  margin-right: 2rem;
}

.closure {
  border-bottom: 1px solid #21BE58;
}
