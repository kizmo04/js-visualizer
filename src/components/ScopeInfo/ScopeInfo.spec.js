import React from "react";
import { shallow, mount, render } from 'enzyme';
import ScopeInfo from "./ScopeInfo";


describe("ScopeInfo", () => {

  const setup = (setupProps = {}) => {
    const defaultProps = {
      scopeHistory: [],
      currentScope: {},
      operationType: ""
    };
    const props = { ...defaultProps, ...setupProps };
    const component = shallow(
      <ScopeInfo
        scopeHistory={props.scopeHistory}
        currentScope={props.currentScope}
        operationType={props.operationType}
      />);

    return {
      props,
      component
    };
  };

  it("renders without crashing", () => {
    const { component } = setup();

    expect(component).toMatchSnapshot();
  });

  describe("with scope props", () => {

    const firstProp = {
      scopeHistory: [],
      currentScope: {
        scopeName: {
          type: "string",
          value: "Global"
        },
        this: {
          type: "string",
          value: "window"
        },
        num: {
          type: "number",
          value: 5
        }
      },
      operationType: "Program"
    };

    it("renders scope name", () => {
      const { component } = setup(firstProp);

      expect(component).toMatchSnapshot();
      expect(component.find('.closure')).to.have.lengthOf(1);
    });
  });
});
