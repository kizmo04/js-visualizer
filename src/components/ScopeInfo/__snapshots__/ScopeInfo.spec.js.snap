// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScopeInfo renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScopeInfo
    currentScope={Object {}}
    operationType=""
    scopeHistory={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="font underline has-text-success operation-type"
        />,
        Array [
          <div
            className="scopeBox"
          />,
        ],
      ],
      "className": "logBox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "font underline has-text-success operation-type",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "scopeBox",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="font underline has-text-success operation-type"
          />,
          Array [
            <div
              className="scopeBox"
            />,
          ],
        ],
        "className": "logBox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "font underline has-text-success operation-type",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "scopeBox",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ScopeInfo with scope props renders scope name 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScopeInfo
    currentScope={
      Object {
        "num": Object {
          "type": "number",
          "value": 5,
        },
        "scopeName": Object {
          "type": "string",
          "value": "Global",
        },
        "this": Object {
          "type": "string",
          "value": "window",
        },
      }
    }
    operationType="Program"
    scopeHistory={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="font underline has-text-success operation-type"
        >
          Program
        </h2>,
        Array [
          <div
            className="scopeBox"
          >
            <h2
              className="font closure has-text-success closure"
            >
              Global
               Closure
            </h2>
            <p
              className="font has-text-success property"
            >
              this: window
            </p>
            <p
              className="font has-text-success property"
            >
              num: 5
            </p>
          </div>,
        ],
      ],
      "className": "logBox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Program",
          "className": "font underline has-text-success operation-type",
        },
        "ref": null,
        "rendered": "Program",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="font closure has-text-success closure"
            >
              Global
               Closure
            </h2>,
            <p
              className="font has-text-success property"
            >
              this: window
            </p>,
            <p
              className="font has-text-success property"
            >
              num: 5
            </p>,
          ],
          "className": "scopeBox",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0scopeName",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Global",
                " Closure",
              ],
              "className": "font closure has-text-success closure",
            },
            "ref": null,
            "rendered": Array [
              "Global",
              " Closure",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": "0this",
            "nodeType": "host",
            "props": Object {
              "children": "this: window",
              "className": "font has-text-success property",
            },
            "ref": null,
            "rendered": "this: window",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "0num",
            "nodeType": "host",
            "props": Object {
              "children": "num: 5",
              "className": "font has-text-success property",
            },
            "ref": null,
            "rendered": "num: 5",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="font underline has-text-success operation-type"
          >
            Program
          </h2>,
          Array [
            <div
              className="scopeBox"
            >
              <h2
                className="font closure has-text-success closure"
              >
                Global
                 Closure
              </h2>
              <p
                className="font has-text-success property"
              >
                this: window
              </p>
              <p
                className="font has-text-success property"
              >
                num: 5
              </p>
            </div>,
          ],
        ],
        "className": "logBox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Program",
            "className": "font underline has-text-success operation-type",
          },
          "ref": null,
          "rendered": "Program",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="font closure has-text-success closure"
              >
                Global
                 Closure
              </h2>,
              <p
                className="font has-text-success property"
              >
                this: window
              </p>,
              <p
                className="font has-text-success property"
              >
                num: 5
              </p>,
            ],
            "className": "scopeBox",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0scopeName",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Global",
                  " Closure",
                ],
                "className": "font closure has-text-success closure",
              },
              "ref": null,
              "rendered": Array [
                "Global",
                " Closure",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": "0this",
              "nodeType": "host",
              "props": Object {
                "children": "this: window",
                "className": "font has-text-success property",
              },
              "ref": null,
              "rendered": "this: window",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "0num",
              "nodeType": "host",
              "props": Object {
                "children": "num: 5",
                "className": "font has-text-success property",
              },
              "ref": null,
              "rendered": "num: 5",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
